cmd/kops/create_cluster.go

```diff
@@ -502,6 +505,32 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 		return fmt.Errorf("Invalid topology %s.", c.Topology)
 	}
 
+	// NAT Gateway/ElasticIP
+	if c.NgwIds != "" {
+		ngwEipList := make([]string, 0)
+		// Perhaps abstract parseZoneList into something more general
+		// But it works for processing comma-delimited strings for now
+		for _, ngwEip := range parseZoneList(c.NgwEips) {
+			ngwEipList = append(ngwEipList, ngwEip)
+		}
+
+		ngwIdList := make([]string, 0)
+		for _, ngwId := range parseZoneList(c.NgwIds) {
+			ngwIdList = append(ngwIdList, ngwId)
+		}
+
+		gatewayIndex := 0
+		for i := range cluster.Spec.Subnets {
+			subnet := &cluster.Spec.Subnets[i]
+			if subnet.Type == api.SubnetTypePrivate {
```

Comment:
Requiring the order to match would be pretty user-unfriendly if we are exposing it to users via a flag, but we're not going to do that (yet).  And if we do, we can rejig this logic (I think we'd have to look up the ngw ids etc)